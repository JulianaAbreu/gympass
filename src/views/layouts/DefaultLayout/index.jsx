import React from 'react';
import { Route, Link } from 'react-router-dom';
import styled from 'styled-components';

import RepositoriesPage from '../../pages/RepositoriesPage';
import CommitsPage from '../../pages/CommitsPage';

import logo from '../../../assets/images/octocat.svg';

const LayoutWrapper = styled.div`
  min-height: 100vh;
  display: grid;
  grid-template-columns: 0px 8fr;
  grid-template-rows: 80px 1fr;
  grid-template-areas:
    'header header header'
    'content content content';
`;

const Content = styled.main`
  grid-area: content;
  padding: 4rem 4rem;
`;

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
  grid-area: header;
  padding: 0 1rem;
  height: 5rem;
  color: white;
  background-color: rgba(0, 0, 0, 0.85);

  span {
    border-bottom: 5px solid #f42b52;
  }
`;

const LogoImg = styled.img`
  width: 2.6rem;
`;

const LogoContent = styled.div`
  display: flex;
  align-items: center;
  h1 {
    margin: 0;
  }
  h1,
  a {
    color: white;
  }
  a {
    margin-left: 30px;
  }
`;

const DefaultLayout = () => (
  <LayoutWrapper>
    <Header>
      <LogoContent>
        <LogoImg src={logo} alt="logo" />
        &nbsp;
        <h1>GitHub</h1>
        <Link to="/">REPOSITORIES</Link>
      </LogoContent>
      <div>
        <span>Hello, reactjs.</span>
      </div>
    </Header>
    <Content>
      <Route path="/" exact component={RepositoriesPage} />
      <Route path="/:repository/commits" exact component={CommitsPage} />
    </Content>
  </LayoutWrapper>
);

export default DefaultLayout;
