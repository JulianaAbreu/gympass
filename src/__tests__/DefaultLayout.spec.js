import React from 'react';
import { Route } from 'react-router-dom';
import { shallow } from 'enzyme';
import styled from 'styled-components';

import DefaultLayout from '../views/layouts/DefaultLayout';

describe('DefaultLayout', () => {
  it('should be a <h1> element in Header styled component', () => {
    const wrapper = shallow(<DefaultLayout />);
    expect(wrapper.find('h1')).toHaveLength(1);
  });

  it('should contain a Route element inside Content', () => {
    const Content = styled.main``;
    const wrapper = shallow(
      <Content>
        <Route />
      </Content>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Route)).toHaveLength(1);
  });

  it('should render a DefaultLayout with content', () => {
    const wrapper = shallow(<DefaultLayout />);
    expect(wrapper.find('div').children()).toHaveLength(1);
  });
});
