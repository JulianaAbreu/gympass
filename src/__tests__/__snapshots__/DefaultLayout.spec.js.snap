// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout should contain a Route element inside Content 1`] = `ShallowWrapper {}`;

exports[`DefaultLayout should contain a route 1`] = `ShallowWrapper {}`;
